---
title: "Lung FACS Notebook"
output:
  html_document: default
  html_notebook: default
---

Enter the directory of the maca folder on your drive and the name of the tissue you want to analyze.

```{r}
tissue_of_interest = "Lung"
```

Load the requisite packages and some additional helper functions.

```{r}
library(here)
library(useful)
library(Seurat)
library(dplyr)
library(Matrix)
library(ontologyIndex)
cell_ontology = get_ontology('https://raw.githubusercontent.com/obophenotype/cell-ontology/master/cl-basic.obo', extract_tags='everything')

validate_cell_ontology = function(cell_ontology_class){
  in_cell_ontology = sapply(cell_ontology_class, function(x) is.element(x, cell_ontology$name) || is.na(x))
  if (!all(in_cell_ontology)) {
    message = paste0('"', cell_ontology_class[!in_cell_ontology], '" is not in the cell ontology
')
    stop(message)
  }
}
convert_to_cell_ontology_id = function(cell_ontology_class){
  return(sapply(cell_ontology_class, function(x) as.vector(cell_ontology$id[cell_ontology$name == x])[1]))
}

save_dir = here('00_data_ingest', 'tissue_robj')
```



```{r}
# read the metadata to get the plates we want
plate_metadata_filename = here('00_data_ingest', '00_facs_raw_data', 'metadata_FACS.csv')

plate_metadata <- read.csv(plate_metadata_filename, sep=",", header = TRUE)
colnames(plate_metadata)[1] <- "plate.barcode"
plate_metadata
```

Subset the metadata on the tissue.

```{r}
tissue_plates = filter(plate_metadata, tissue == tissue_of_interest)[,c('plate.barcode','tissue','subtissue','mouse.sex')]
tissue_plates
```

Load the read count data.
```{r}
#Load the gene names and set the metadata columns by opening the first file
filename = here('00_data_ingest', '00_facs_raw_data', 'FACS', paste0(tissue_of_interest, '-counts.csv'))

raw.data = read.csv(filename, sep=",", row.names=1)
# raw.data = data.frame(row.names = rownames(raw.data))
corner(raw.data)
```
Make a vector of plate barcodes for each cell

```{r}
plate.barcodes = lapply(colnames(raw.data), function(x) strsplit(strsplit(x, "_")[[1]][1], '.', fixed=TRUE)[[1]][2])
head(plate.barcodes)
```

Use only the metadata rows corresponding to Bladder plates. Make a plate barcode dataframe to "expand" the per-plate metadata to be per-cell.
```{r}
barcode.df = t.data.frame(as.data.frame(plate.barcodes))

rownames(barcode.df) = colnames(raw.data)
colnames(barcode.df) = c('plate.barcode')
head(barcode.df)

rnames = row.names(barcode.df)
meta.data <- merge(barcode.df, plate_metadata, by='plate.barcode', sort = F)
row.names(meta.data) <- rnames

# Sort cells by plate barcode because that's how the data was originally
meta.data = meta.data[order(meta.data$plate.barcode), ]
corner(meta.data)
raw.data = raw.data[, rownames(meta.data)]
corner(raw.data)
```



```{r}

erccs <- grep(pattern = "^ERCC-", x = rownames(x = raw.data), value = TRUE)
percent.ercc <- Matrix::colSums(raw.data[erccs, ])/Matrix::colSums(raw.data)
ercc.index <- grep(pattern = "^ERCC-", x = rownames(x = raw.data), value = FALSE)
raw.data <- raw.data[-ercc.index,]

tiss <- CreateSeuratObject(raw.data = raw.data, project = tissue_of_interest, min.cells = 5, min.genes = 5)

tiss <- AddMetaData(object = tiss, meta.data)
tiss <- AddMetaData(object = tiss, percent.ercc, col.name = "percent.ercc")
# Change default name for sums of counts from nUMI to nReads
colnames(tiss@meta.data)[colnames(tiss@meta.data) == 'nUMI'] <- 'nReads'
```


Calculate percent ribosomal genes.

```{r}
ribo.genes <- grep(pattern = "^Rp[sl][[:digit:]]", x = rownames(x = tiss@data), value = TRUE)

percent.ribo <- Matrix::colSums(tiss@raw.data[ribo.genes, ])/Matrix::colSums(tiss@raw.data)

tiss <- AddMetaData(object = tiss, metadata = percent.ribo, col.name = "percent.ribo")
```

A sanity check: reads vs genes.

```{r}
GenePlot(object = tiss, gene1 = "nReads", gene2 = "nGene")
```

Filter out cells with few reads and few genes.

```{r}
tiss <- FilterCells(object = tiss, subset.names = c("nGene", "nReads"), 
    low.thresholds = c(500, 50000), high.thresholds = c(25000, 2000000))
```


Normalize the data, then regress out correlation with total reads
```{r}
tiss <- NormalizeData(object = tiss, scale.factor = 1e6)
tiss <- ScaleData(object = tiss)
tiss <- FindVariableGenes(object = tiss, do.plot = TRUE, x.high.cutoff = Inf, y.cutoff = 0.4, x.low.cutoff = 0.75)
```


Run Principal Component Analysis.
```{r}
tiss <- RunPCA(object = tiss, pc.genes = tiss@var.genes, do.print = TRUE, pcs.print = 1:5, genes.print = 5, pcs.compute = 30)
```

```{r}
tiss <- ProjectPCA(object = tiss, do.print = FALSE)
```


```{r, echo=FALSE, fig.height=4, fig.width=8}
PCHeatmap(object = tiss, pc.use = 18:30, cells.use = 500, do.balanced = TRUE, label.columns = FALSE, num.genes = 8)
```

```{r}
PCElbowPlot(object = tiss)
```


```{r}
n.pcs = 20
tiss <- FindClusters(object = tiss, reduction.type = "pca", dims.use = 1:n.pcs, resolution = 3, print.output = 0, save.SNN = TRUE, force.recalc = TRUE)
```


```{r}
tiss <- RunTSNE(object = tiss, dims.use = 1:n.pcs, seed.use = 10, check_duplicates = F)
```

```{r}
# note that you can set do.label=T to help label individual clusters
TSNEPlot(object = tiss, do.label = T, no.axes=T, pt.size = 1)
```

Check expression of genes of interset.
```{r}
genes_to_check = c('Pecam1', 'Epcam', 'Col1a1', 'Ptprc')

FeaturePlot(tiss, genes_to_check, pt.size = 1, cols.use = c('grey', 'red'))
```

```{r}
genes_to_check = c('Calca', 'Resp18', 'Ager', 'Scgb1a1')

FeaturePlot(tiss, genes_to_check, pt.size = 1, cols.use = c('grey', 'red'))
```

```{r}
genes_to_check = c('Foxj1', 'Cd79a', 'Klrb1c', 'Cd3e')

FeaturePlot(tiss, genes_to_check, pt.size = 1, cols.use = c('grey', 'red'))
```

```{r}
genes_to_check = c('H2-Aa', 'Lgals3', 'Itgal', 'Csf1r', 'Msln', 'Chad')

FeaturePlot(tiss, genes_to_check, pt.size = 1, cols.use = c('grey', 'red'))
```


Color by metadata, like plate barcode, to check for batch effects.
```{r}
TSNEPlot(object = tiss, do.return = TRUE, group.by = "plate.barcode")
```

Find differentially expressed markers.
```{r}
tiss.markers <- FindAllMarkers(object = tiss, only.pos = TRUE, min.pct = 0.25, 
    thresh.use = 0.25)
```

Display top 4 markers per cluster.
```{r}
tiss.markers %>% group_by(cluster) %>% top_n(4, avg_diff)
```


```{r}
# To change the y-axis to show raw counts, add use.raw = T.
VlnPlot(tiss, genes_to_check)
```


```{r}
table(tiss@ident)
```


```{r}
tiss = BuildClusterTree(tiss)
```

Get markers for a particular cluster
```{r}
cluster_markers = filter(tiss.markers, cluster == 4)$gene

DotPlot(tiss, genes.plot = cluster_markers[1:10])
```


## Assigning cell type identity to clusters

At a coarse level, we can use canonical markers to match the unbiased clustering to known cell types:

```{r}
# stash current cluster IDs
tiss <- StashIdent(object = tiss, save.name = "cluster.ids")

# enumerate current cluster IDs and the labels for them
cluster.ids <- c(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20)
cell_ontology_class <-
  c(
  "Endothelial Cell Type I",
  "Stromal Cell Type I",
  "Endothelial Cell Type I",
  "Stromal Cell Type II &\n Mesothelial Cells",
  "Endothelial Cell Type I",
  "Aveolar Epithelial Cell Type II",
  "Natural Killer & T Cells",
  "Endothelial Cell Type II",
  "Immune Cell Type I",
  "Endothelial Cell Type III",
  'Immune Cell Type II',
  'Endothelial Cell Type IV',
  'Immune Cell Type III',
  'B cell',
  'Stromal Cell Type III',
  'Endothelial Cell Type II',
  'Stromal Cell Type IV',
  'Endothelial Cell Type IV',
  'Ciliated, Club, Neuroendocine,\n and Alveolar Epithelial Type I Cells',
  'Immune Cell Type IV',
  'Stromal Cell Type V'
  )

tiss@meta.data[,'cell_ontology_class'] <- plyr::mapvalues(x = tiss@meta.data$cluster.ids, from = cluster.ids, to = cell_ontology_class)

TSNEPlot(object = tiss, do.label = TRUE, pt.size = 0.5, group.by='cell_ontology_class', no.legend = TRUE, no.axes = TRUE)
```


## Checking for batch effects


```{r}
TSNEPlot(object = tiss, do.return = TRUE, group.by = "mouse.sex")
```

Print a table showing the count of cells in each identity category from each plate.

```{r}
table(as.character(tiss@ident), as.character(tiss@meta.data$plate.barcode))
```


## Subset and iterate

We can repeat the above analysis on a subset of genes, defined using cluster IDs or expression or some other metadata. This is a good way to drill down and find substructure.

```{r}
# Get subtiss, eg, the endocrine clusters
subtiss <- SubsetData(object = tiss, ident.use = c(0, 1, 12),
  do.center = F, do.scale = F)

subtiss <- NormalizeData(object = subtiss)
subtiss@scale.data = subtiss@data
#subtiss <- ScaleData(object = subtiss)
#subtiss <- ScaleData(object = subtiss, vars.to.regress = c("nReads", "percent.ribo","Rn45s"))
```



```{r}
sub.n.pcs = 20
subtiss <- FindVariableGenes(object = subtiss, do.plot = TRUE, x.high.cutoff = Inf, y.cutoff = 0.5)
subtiss <- RunPCA(object = subtiss, pcs.compute = sub.n.pcs)

subtiss <- ProjectPCA(object = subtiss, do.print = FALSE)
```


```{r}
PCElbowPlot(object = subtiss)
```

```{r}
PCHeatmap(object = subtiss, pc.use = 1:6, cells.use = 500, do.balanced = TRUE, label.columns = FALSE, num.genes = 12)
```


```{r}
subtiss <- FindClusters(object = subtiss, reduction.type = "pca", dims.use = 1:sub.n.pcs, 
    resolution = 1.0, print.output = 0, save.SNN = TRUE, force.recalc = T)
```


```{r}
subtiss <- RunTSNE(object = subtiss, dims.use = 1:sub.n.pcs, seed.use = 10, check_duplicates = F)
```

```{r}
# note that you can set do.label=T to help label individual clusters
TSNEPlot(object = subtiss, do.label = T)
```


```{r}
write.csv(tiss@meta.data[,c('cell_ontology_class')],file =paste0(save_dir,"/", "SmartSeq2_", tissue_of_interest,"_cell_ontology_class.csv"))
```
