---
 title: "Muscle Notebook"
 output: html_notebook
---

Specify the tissue of interest, run the boilerplate code which sets up the functions and environment, load the tissue object.

```{r}
tissue_of_interest = "Muscle"
library(here)
source(here("00_data_ingest", "02_tissue_analysis_rmd", "boilerplate.R"))
tiss <- load_tissue_droplet(tissue_of_interest)
```

Visualize top genes in principal components

```{r, echo=FALSE, fig.height=4, fig.width=8}
PCHeatmap(object = tiss, pc.use = 1:3, cells.use = 500, do.balanced = TRUE, label.columns = FALSE, num.genes = 8)
```

Later on (in FindClusters and TSNE) you will pick a number of principal components to use. This has the effect of keeping the major directions of variation in the data and, ideally, supressing noise. There is no correct answer to the number to use, but a decent rule of thumb is to go until the plot plateaus.

```{r}
PCElbowPlot(object = tiss)
```

Choose the number of principal components to use.
```{r}
# Set number of principal components. 
n.pcs = 15
```

The clustering is performed based on a nearest neighbors graph. Cells that have similar expression will be joined together. The Louvain algorithm looks for groups of cells with high modularity--more connections within the group than between groups. The resolution parameter determines the scale. Higher resolution will give more clusters, lower resolution will give fewer.

For the top-level clustering, aim to under-cluster instead of over-cluster. It will be easy to subset groups and further analyze them below.

```{r}
# Set resolution 
res.used <- 1

tiss <- FindClusters(object = tiss, reduction.type = "pca", dims.use = 1:n.pcs, 
    resolution = res.used, print.output = 0, save.SNN = TRUE)
```

We use TSNE solely to visualize the data.
```{r}
# If cells are too spread out, you can raise the perplexity. If you have few cells, try a lower perplexity (but never less than 10).
tiss <- RunTSNE(object = tiss, dims.use = 1:n.pcs, seed.use = 10, perplexity=30)
```

```{r}
TSNEPlot(object = tiss, do.label = T, pt.size = 1.2, label.size = 4)
```


```{r}
genes_to_check = c('Myf5', 'Myod1', 'Pdgfra', 'Atxn1', 'Pecam1','Ptprc','Itgam','Cd3g','Cd4', 'Cd19', 'Chad','Chodl', 'Sox9', 'Acan','Acta2')
```

```{r, echo=FALSE, fig.height=4, fig.width=12}
FeaturePlot(tiss, genes_to_check, pt.size = 1, nCol = 5, cols.use = c("grey", "red"))
```

Dotplots let you see the intensity of exppression and the fraction of cells expressing for each of your genes of interest.
The radius shows you the percent of cells in that cluster with at least one read sequenced from that gene. The color level indicates the average
Z-score of gene expression for cells in that cluster, where the scaling is done over taken over all cells in the sample.

```{r, echo=FALSE, fig.height=8, fig.width=10}
DotPlot(tiss, genes_to_check, plot.legend = T, col.max = 2.5, do.return = T) + coord_flip()
```

```{r, fig.height = 8, fig.width = 8}
VlnPlot(tiss, "Myf5", use.raw= T, nCol = 2)
```


We can also find all differentially expressed genes marking each cluster. This may take some time.

```{r}
#clust.markers0 <- FindMarkers(object = tiss, ident.1 = 0, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
#tiss.markers <- FindAllMarkers(object = tiss, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
```

Display the top markers you computed above.
```{r}
#tiss.markers %>% group_by(cluster) %>% top_n(5, avg_diff)
```

Using the markers above, we can confidentaly label many of the clusters.

```{r}
tiss <- StashIdent(object = tiss, save.name = "cluster.ids")

cluster.ids <- c(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14)

cluster.ids <- c(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14)
free_annotation <-
  c(
  "mesenchymal stem cell",
  "endothelial cell",
  "B cell",
  "chondrocyte-like",
  "endothelial cell",
  "skeletal muscle satellite cell",
  "T cell",
  "macrophage",
  "mesenchymal stem cell",
  "mesenchymal stem cell",
  "endothelial cell",
  "endothelial cell",
  "unknown",
  "unknown",
  "unknown")

cell_ontology_class <-
  c(
  "mesenchymal stem cell",
  "endothelial cell",
  "B cell",
  NA,
  "endothelial cell",
  "skeletal muscle satellite cell",
  "T cell",
  "macrophage",
  "mesenchymal stem cell",
  "mesenchymal stem cell",
  "endothelial cell",
  "endothelial cell",
  NA,
  NA,
  NA)

validate_cell_ontology(cell_ontology_class)
cell_ontology_id = convert_to_cell_ontology_id(cell_ontology_class)

tiss@meta.data['cell_ontology_class'] <- as.character(plyr::mapvalues(x = tiss@ident, from = cluster.ids, to = cell_ontology_class))
validate_cell_ontology(cell_ontology_class)
cell_ontology_id = convert_to_cell_ontology_id(cell_ontology_class)

tiss@meta.data['free_annotation'] <- as.character(plyr::mapvalues(x = tiss@ident, from = cluster.ids, to = free_annotation))
tiss@meta.data['cell_ontology_id'] <- as.character(plyr::mapvalues(x = tiss@ident, from = cluster.ids, to = cell_ontology_id))
```

## Checking for batch effects

Color by metadata, like plate barcode, to check for batch effects.
```{r}
TSNEPlot(object = tiss, do.return = TRUE, group.by = "channel")
```


# Final coloring

Color by cell ontology class on the original TSNE.

```{r}
TSNEPlot(object = tiss, do.return = TRUE, group.by = "cell_ontology_class")
```

# Save the Robject for later

```{r}
filename = here('00_data_ingest', '04_tissue_robj_generated', 
                     paste0("droplet_", tissue_of_interest, "_seurat_tiss.Robj"))
print(filename)
save(tiss, file=filename)
```

```{r}
# To reload a saved object
# filename = here('00_data_ingest', '04_tissue_robj_generated', 
#                      paste0("droplet_", tissue_of_interest, "_seurat_tiss.Robj"))
# load(file=filename)
```


# Export the final metadata

So that Biohub can easily combine all your cell_ontology_classs, please export them as a simple csv.

```{r}
head(tiss@meta.data)
```

```{r}
filename = here('00_data_ingest', '03_tissue_annotation_csv', 
                     paste0(tissue_of_interest, "_droplet_annotation.csv"))
write.csv(FetchData(tiss, c('channel','cell_ontology_class','cell_ontology_id', 'free_annotation', 'tSNE_1', 'tSNE_2')), file=filename)
```
Figures for Supplement: Dot plot above is sufficient and code is copied below.
```{r, echo=FALSE, fig.height=8, fig.width=10}
DotPlot(tiss, genes_to_check, plot.legend = T, col.max = 2.5, do.return = T) + coord_flip()
```
```


Write the cell ontology and free annotations to CSV.

```{r}
filename = here('00_data_ingest', '03_tissue_annotation_csv', 
                    paste0(tissue_of_interest, "_droplet_annotation.csv"))
write.csv(tiss@meta.data[,c('channel,'cell_ontology_class','cell_ontology_id', 'free.annotation')], file=filename)
```
