---
title: "Tissue Annotation Vignette"
output:
  pdf_document: default
  html_notebook: default
---

```{r, include = FALSE}
library(here)
source(here("00_data_ingest", "02_tissue_analysis_rmd", "boilerplate.R"))
```

## Preprocess

Enter the name of the tissue you want to analyze.

```{r}
tissue_of_interest = "Liver"
```

```{r, include=FALSE}
# Load the per-plate metadata
plate_metadata_filename = here('00_data_ingest', '00_facs_raw_data', 'metadata_FACS.csv')

plate_metadata <- read.csv(plate_metadata_filename, sep=",", header = TRUE)
colnames(plate_metadata)[1] <- "plate.barcode"


# Load the gene names and set the metadata columns by opening the first file
filename = here('00_data_ingest', '00_facs_raw_data', 'FACS', paste0(tissue_of_interest, '-counts.csv'))

raw.data = read.csv(filename, sep=",", row.names=1)

plate.barcodes = lapply(colnames(raw.data), function(x) strsplit(strsplit(x, "_")[[1]][1], '.', fixed=TRUE)[[1]][2])

barcode.df = t.data.frame(as.data.frame(plate.barcodes))

rownames(barcode.df) = colnames(raw.data)
colnames(barcode.df) = c('plate.barcode')

rnames = row.names(barcode.df)
meta.data <- merge(barcode.df, plate_metadata, by='plate.barcode', sort = F)
row.names(meta.data) <- rnames

# Sort cells by cell name
meta.data = meta.data[order(rownames(meta.data)), ]
raw.data = raw.data[, rownames(meta.data)]

# Find ERCC's, compute the percent ERCC, and drop them from the raw data.
erccs <- grep(pattern = "^ERCC-", x = rownames(x = raw.data), value = TRUE)
percent.ercc <- Matrix::colSums(raw.data[erccs, ])/Matrix::colSums(raw.data)
ercc.index <- grep(pattern = "^ERCC-", x = rownames(x = raw.data), value = FALSE)
raw.data <- raw.data[-ercc.index,]

# Create the Seurat object with all the data
tiss <- CreateSeuratObject(raw.data = raw.data, project = tissue_of_interest)
tiss <- AddMetaData(object = tiss, meta.data)
tiss <- AddMetaData(object = tiss, percent.ercc, col.name = "percent.ercc")

# Change default name for sums of counts from nUMI to nReads
colnames(tiss@meta.data)[colnames(tiss@meta.data) == 'nUMI'] <- 'nReads'
  
# Create metadata columns for annotations
tiss@meta.data[,'free_annotation'] <- NA
tiss@meta.data[,'cell_ontology_class'] <- NA
```

We begin by loading a gene by cell matrix $A$, where $A_{ij}$ is the number of reads (or UMIs) from cell $j$ that aligned to gene $i$. The Seurat package keeps track of matrices and coordinates derived from $A$ together with metadata; that information is stored in a large object called `tiss`. For the sake of explicitness, we describe below many of the mathematical transformations implemented inside of Seurat.

First, we filter out cells with fewer than 500 genes or 50,000 reads. (For UMI data, we filter out cells with fewer than 500 genes or 1,000 UMIs.)

```{r}
tiss <- FilterCells(object = tiss, subset.names = c("nGene", "nReads"), 
                    low.thresholds = c(500, 50000))
```

Next, we log-normalize counts for each cell, essentially to log counts per million (for reads) or counts per ten thousand (UMIs). More precisely, we set

$$N_{ij} = \log \left (1 + M \frac{A_{ij}}{\sum_{j^\prime} A_{i j^\prime}} \right ),$$
where $M = 10^6$ for FACS and $M = 10^4$ for droplets. The log is base $e$. We chose those values of $M$ to be close to the average number of counts per cell.

```{r}
tiss <- NormalizeData(object = tiss, scale.factor = 1e6)
```

Then we shift and scale the rows of the normalized matrix, so each gene has mean zero and variance one.
$$X_{ij} = (N_{ij} - \mu_i)/\sigma_i,$$
where $\mu_i$ is the mean of $N_{ij}$ and $\sigma_i$ is the standard deviation of $N_{ij}$.

```{r, warning = FALSE, message=FALSE}
tiss <- ScaleData(object = tiss)
```

We select variable genes (those with high standardized log dispersion, given their mean), as they will be more informative about differences between cell types in the population.

Concretely, the log dispersion $d_i$ of a gene $i$ is $d_i = \log(v_i/m_i),$ where $v_i$ is the variance of $\exp(N_{ij})$ and $m_i$ is the mean of $\exp(N_{ij})$. (In other words, $m_i$ is $1$ plus CPM.) We bin the genes into 20 equal-spaced bins based on $\log(m_i)$, then compute the mean and standard deviation of $d_i$ within each bin. The standardized log dispersion $\bar{d}_i$ is the dispersion $d_i$ shifted by the mean and rescaled by the standard deviation of the $d_k$ within its bin. We retain genes with $\bar{d}_i > 0.5$ and $\log(m_i) > 0.1$.

```{r, warning = FALSE, message = FALSE}
tiss <- FindVariableGenes(object = tiss, do.plot = TRUE, 
                          x.high.cutoff = Inf, y.cutoff = 0.5, x.low.cutoff = 0.1)
```

Cells are projected onto a low-dimensional subspace using principal component analysis on the scaled expression $X$ of the variable genes.

```{r}
tiss <- RunPCA(object = tiss, do.print = FALSE)
tiss <- ProjectPCA(object = tiss, do.print = FALSE)
```

We can visualize top genes in each principal component.

```{r, echo=FALSE}
PCHeatmap(object = tiss, pc.use = 1:3, cells.use = 500, do.balanced = TRUE, label.columns = FALSE, num.genes = 8)
```

We then project onto just the top principal components. This has the effect of keeping the major directions of variation in the data and, ideally, supressing noise. A decent rule of thumb is to pick the elbow in the plot below.

```{r}
PCElbowPlot(object = tiss)
```

Choose the number of principal components to use.

```{r}
n.pcs = 11
```

## Cluster

The clustering is performed on a shared-nearest-neighbors graph on the cells. 

The shared-nearest-neighbors graph was constructed based on the Euclidean distance in the low-dimensional subspace; cells are connected if their $k$-neighborhoods overlap. Indeed, let $\mathcal{N}(j)$ denote the k-nearest neighborhood (k = 30) for a cell $j$. The shared-nearest-neighbor graph G has a vertex for each cell and an edge of weight
$$w_{jk} = \frac{
\vert \mathcal{N}(j) \cap \mathcal{N}(k) \vert}{\vert \mathcal{N}(j) \cup \mathcal{N}(k) \vert}$$
between cells $j$ and $k$.
Cells were clustered using a [modified version](https://journals.aps.org/pre/abstract/10.1103/PhysRevE.74.016110) of the Louvain method for modularity maximization. The modularity has a resolution parameter $\gamma$,

$$Q = \sum_{ij} \left (A_{ij} - \gamma \frac{k_i k_j}{2m} \right ) \delta(c_i, c_j),$$

where $A_{ij}$ is the weighted adjacency matrix, $k_i$ and $k_j$ are the weighted degrees of cells $i$ and $j$, $m$ is the total weight of edges in the graph, $c_i$ denotes cluster membership, and $\delta(c_i,c_j)$ is $1$ if $i$ and $j$ are in the same cluster, and $0$ otherwise.

The resolution $\gamma$ is a tuneable parameter in this analysis that sets the tradeoff between in-group connections and between-group connections. High resolution favors smaller clusters.

```{r}
# Set resolution 
res.used <- 1

tiss <- FindClusters(object = tiss, reduction.type = "pca", dims.use = 1:n.pcs, 
    resolution = res.used, print.output = 0, save.SNN = TRUE)
```

We use tSNE solely to visualize the data.

```{r}
tiss <- RunTSNE(object = tiss, dims.use = 1:n.pcs, seed.use = 10, perplexity=30)
```

```{r}
TSNEPlot(object = tiss, do.label = T, pt.size = 1.2, label.size = 4)
```

## Label clusters using marker genes

Check expression of genes useful for indicating cell type.

```{r}
genes_hep = c('Alb', 'Ttr', 'Apoa1', 'Serpina1c') #hepatocyte
genes_endo = c('Pecam1', 'Nrp1', 'Kdr','Oit3') # endothelial
genes_kuppfer = c('Emr1', 'Clec4f', 'Cd68', 'Irf7') # Kuppfer cells
genes_nk = c('Zap70', 'Il2rb', 'Nkg7', 'Cxcr6') # Natural Killer cells
genes_b = c('Cd79a', 'Cd79b', 'Cd74', 'Cd19') # B Cells

genes_all = c(genes_hep, genes_endo, genes_kuppfer, genes_nk, genes_b)
```

In the tSNE plots below, the intensity of each point represents the log-normalized gene expression $N_{ij}$.

```{r, echo=FALSE, fig.height=20, fig.width=16}
FeaturePlot(tiss, genes_all, pt.size = 3, nCol = 4, cols.use = c("lightgrey", "blue"), no.legend = F)
```

Dotplots show, for each cluster and gene, the fraction of cells with at least one read for the gene (circle size) and the average scaled expression for that gene among the cells expressing it (circle color).

```{r, echo=FALSE}
DotPlot(tiss, genes_all, plot.legend = T, col.max = 2.5, x.lab.rot = T)
```

The low but nonzero levels of Albumin present in all clusters is consistent with a small amount of leakage, either through physical contamination or index hopping. Nevertheless, the absolute levels of expression confirm a sharp difference between the hepatocyte clusters and the others.

```{r, echo=FALSE, fig.height=3, fig.width=6}
VlnPlot(tiss, 'Alb', use.raw = T, do.return = T)
```

To confirm the identity of a cluster, you can inspect the genes differentially expressed in that cluster compared to the others.

```{r}
clust.markers7 <- FindMarkers(object = tiss, ident.1 = 7, 
                              only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
```

The top markers for cluster 7 include histocompatibility markers H2-*, consistent with the expression of other B-cell markers seen above.

```{r}
head(clust.markers7)
```

Using the markers, we can confidentaly label the clusters. We provide both a free annotation (where any name can be used) and a cell ontology class. The latter uses a controlled vocabulary for easy comparison between studies and different levels of the taxonomy.

```{r}
tiss <- StashIdent(object = tiss, save.name = "cluster.ids")

cluster.ids <- c(0, 1, 2, 3, 4, 5, 6, 7, 8)

free_annotation <- c(
  "endothelial cell",
  "hepatocyte",
  "hepatocyte",
  "hepatocyte",
  "hepatocyte",
  "kuppfer",
  "hepatocyte",
  "B cell",
  "NK/NKT cells")

cell_ontology_class <-c(
  "endothelial cell of hepatic sinusoid",
  "hepatocyte",
  "hepatocyte",
  "hepatocyte",
  "hepatocyte",
  "Kupffer cell",
  "hepatocyte",
  "B cell",
  "natural killer cell")

tiss = stash_annotations(tiss, cluster.ids, free_annotation, cell_ontology_class)
```


## Checking for batch effects

Color by metadata, like plate barcode, to check for batch effects. Here we see that the clusters are segregated by sex.

```{r}
TSNEPlot(object = tiss, do.return = TRUE, group.by = "mouse.id")
```

Nevertheless, every cluster contains cells from multiple mice.

```{r}
table(FetchData(tiss, c('mouse.id','ident')) %>% droplevels())
```

# Final coloring

Color by cell ontology class on the original tSNE.

```{r}
TSNEPlot(object = tiss, group.by = "cell_ontology_class")
```


# Save the Robject for later

```{r}
#filename = here('00_data_ingest', '04_tissue_robj_generated', 
#                     paste0("facs_", tissue_of_interest, "_seurat_tiss.Robj"))
#print(filename)
#save(tiss, file=filename)
```

```{r}
# To reload a saved object
#filename = here('00_data_ingest', '04_tissue_robj_generated',
#                      paste0("facs_", tissue_of_interest, "_seurat_subtiss.Robj"))
#load(file=filename)
```

# Export the final metadata

```{r}
#save_annotation_csv(tiss, tissue_of_interest, "facs")
```


