---
title: "R Notebook"
output: html_notebook
---

# Load data and metadata
```{r}
rm(list=ls())
require(Seurat)
require(dplyr)
require(data.table)
require(Hmisc)
require(doMC)
library(here)
library(dplyr)
library(tidyr)
library(readr)


set.seed(644685)

```


```{r load tabula muris data}
tm.facs.matrix = readRDS(here("data-tabula-muris", "TM_facs_mat.rds"))
rownames(tm.facs.matrix) <- make.names(rownames(tm.facs.matrix))
tm.facs.metadata = read_csv(here("data-tabula-muris", "TM_facs_metadata.csv"))

tm.facs.metadata <- tm.facs.metadata %>% filter(!is.na(tissue), !is.na(cell_ontology_class), !is.na(cell_ontology_id))
tm.facs.metadata <- tm.facs.metadata %>% filter(cell %in% colnames(tm.facs.matrix))

tm.facs.metadata$tissue <- make.names(tm.facs.metadata$tissue)
tm.facs.metadata$cell_ontology_class <- make.names(tm.facs.metadata$cell_ontology_class)
# make "annotation.2"
tm.facs.metadata  <- tm.facs.metadata %>% mutate(annotation.2 = paste0(tissue, "__", cell_ontology_class))

tissue_colors <- read_csv(here('00_data_ingest/15_color_palette/tissue_colors.csv'))
colnames(tissue_colors) <- c('tissue','tiss.color')
tissue_colors$tissue <- make.names(tissue_colors$tissue)

tm.facs.metadata <- inner_join(tm.facs.metadata, tissue_colors, by = 'tissue')

tm.facs.matrix <- tm.facs.matrix[, tm.facs.metadata$cell]

```


```{r load TFs}
# (1140 genes from MGI, filtered by GO Term = "DNA binding transcription factor activity", GO ID ?)
tfs      <- read.csv(here('23_tf_analysis','GO_term_summary_20171110_222852.csv'))
tf.names <- as.character( tfs %>% distinct(Symbol) %>% pull(Symbol) )
tf.names <- make.names(tf.names)
tf.names <- tf.names[tf.names %in% rownames(tm.facs.matrix)]
length(tf.names)

# # remove genes that are not detected
# x=apply(as.matrix(tissFACS@data[tf.names, ]) > 0, 1, sum) 
# genes.expr <- names(x[x > 5])
# tf.names <- tf.names[tf.names %in% genes.expr]

# Remove dissociation-related genes
iegs <- as.character(read.csv(here('00_data_ingest','20_dissociation_genes','genes_affected_by_dissociation_unix.csv'), header = T)[,1])
iegs <- iegs[!iegs=="Cebpb"]

tf.names <- tf.names[!tf.names %in% iegs]
length(tf.names)
```

```{r prepare data for TF analysis}
rownames(tm.facs.metadata) = tm.facs.metadata$cell
tissFACS <- CreateSeuratObject(raw.data = tm.facs.matrix, meta.data = tm.facs.metadata)
tissFACS <- NormalizeData(tissFACS, scale.factor = 1e6)
```


# Functions: average and fraction expressing per group

```{r}
group_averages <- function(mat, groups){
  group_names = unique(groups)
  means = matrix(0, dim(mat)[1], length(group_names))
  colnames(means) = group_names
  rownames(means) = rownames(mat)
  for(group in group_names){
    means[,group] = Matrix::rowMeans(mat[,groups == group,drop=FALSE])
  }
  means
}

# thresh = 1 is a cutoff of ~2 reads per million in ln(1+cpm) space
group_fractions <- function(mat, groups, thresh=1){
  group_names = unique(groups)
  fractions = matrix(0, dim(mat)[1], length(group_names))
  colnames(fractions) = group_names
  rownames(fractions) = rownames(mat)
  mat.bin <- mat > thresh
  for(group in group_names){
    fractions[,group] = Matrix::rowMeans(mat.bin[,groups == group,drop=FALSE])
  }
  fractions
}

```


# Calculate fractions
```{r}
tissFACS <- SetAllIdent(tissFACS, 'annotation.2')

fractions <- group_fractions(tissFACS@data[tf.names, ], tissFACS@ident)
write.csv(fractions, "fractions.csv")
```

# Analyze cell type specificity of genes from known TF reprogramming protocols
```{r}
fractions <- read.csv("fractions.csv", row.names = 1)
```


```{r}
# A.k.a "BAM"
print("Target cell type: neuron")
thresh=0.1
genes <- c("Pou3f2","Ascl1","Myt1l")
colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]

# A.k.a "BAZ"
print("Target cell type: neuron")
thresh=.1
genes <- c("Pou3f2","Ascl1","Zic1")
colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]

# Classic Cardiomyocyte protocol
print("Target cell type: Cardiomyocyte")
thresh=.2
genes <- c("Gata4","Tbx5","Mef2c")
colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]

# More recent Cardiomyocyte differentiation (https://doi.org/10.1016/j.yjmcc.2013.04.004)
print("Target cell type: Cardiomyocyte")
thresh=.2
genes <-c("Hand2", "Nkx2.5", "Gata4", "Mef2c", "Tbx5")
colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]

# Liver results: none of the protocols uniquely specify liver
# Liver hepatocyte protocol (although only a single Fox gene was used in first publication; no difference observed btwn all 3)
print("Target cell type: Hepatocyte")
thresh=.3
genes <- c("Hnf4a","Foxa1","Foxa2","Foxa3")
colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]

# Mouse hepatocyte differentiation from fibroblasts (doi:10.1038/nature10116)
print("Target cell type: Hepatocyte")
thresh=.2
genes <- c("Hnf1a","Gata4","Foxa3")
colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]


# Other liver hepatocyte protocol
print("Target cell type: Hepatocyte")
thresh=.4
genes <- c("Hnf4a","Cebpb","Foxa2")
genes <- genes[genes %in% rownames(fractions)]

colnames(fractions)[colSums(fractions[genes, ] > thresh) == length(genes)]

# Note: Pou5f1 is not detected in any cells even though it is in the annotation
# This makes sense as we do not have any pluripotent stem cells annotated in the data
print("Target cell type: Pluripotent stem cell")
thresh=.2
genes <- c("Myc","Sox2","Klf4","Pou5f1")
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]

# NSCs are annotated in the free_annotation group but not in the cell ontology, so this will be excluded for now
# print("Target cell type: Neural stem cell")
# thresh=.05
# genes <- c("Myc","Sox2","Klf4","Tcf3","Pou3f4")
# colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]

# The first direct reprogramming protocol from '87 (doi:10.1016/0092-8674(87)90585-X)
print("Target cell type: Myoblast")
thresh=.3
genes <- c("Myod1")
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]

# TFs for MEF-Oligodendrocyte reprogramming (10.1038/nbt.2564)
print("Target cell type: Oligodendrocyte")
thresh=.3
genes <- c("Sox10","Olig2","Zfp536")
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]


# 10.1016/j.stem.2017.08.020
# Target cell type: fetal intestinal progenitor cells
genes <- c("Hnf4a","Foxa3","Gata6","Cdx2")
genes <- genes[genes %in% rownames(fractions)]
print(genes)
thresh=.4
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]

# target cell type: hematopoeitic progenitors
# 10.1016/j.celrep.2014.11.002
genes <- c("Erg","Gata2","Lmo2","Runx1","Tal1")
genes <- genes[genes %in% rownames(fractions)]
print(genes)
thresh=.4
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]


# TFs that turn B cells into macrophages (doi:10.1016/S0092-8674(04)00419-2)
print("Target cell type: Macrophage (from B cells)")
thresh=.8
genes <- c("Cebpa","Cebpb")
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]


# TFs that turn exocrine cells into beta cells (in vivo reprogramming) (doi:10.1038/nature07314)
print("Target cell type: Beta cell (in vivo, from exocrine cells)")
thresh=.05
genes <- c("Neurog3","Pdx1","Mafa")
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]
```

Human cell-specific conversion protocols
```{r}

# TFs for conversion of human cells to endothelial cells (doi:10.1073/pnas.1413234112)
print("Target cell type: human endothelial cells")
thresh=.05
genes <- c("Etv2")
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]

# TFs for conversion of human cells to osteoblasts (10.1073/pnas.1420713112)
print("Target cell type: human osteoblasts")
thresh=.01
genes <- c("Runx2","Sp7","Mycl1")
# genes[!genes %in% rownames(fractions)]
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]

# TFs for conversion of human cells to osteoblasts (10.1073/pnas.1420713112)
print("Target cell type: human osteoblasts")
thresh=.01
genes <- c("Pou5f1")
# genes[!genes %in% rownames(fractions)]
colnames(fractions)[colSums(fractions[genes, ,drop=F] > thresh) == length(genes)]


```


# Output a table for the paper
```{r}
known.protocols <- read_csv("known_reprogramming_protocols.csv")
celltype.translate <- read_csv("translate_celltype_cellOntology.csv")
celltype.translate <- distinct(celltype.translate)
known.protocols <- known.protocols %>% filter(Genes != "", !is.na(Fraction_Expressing_Threshold))
known.protocols <- inner_join(known.protocols, celltype.translate, by="CellType")
output <- known.protocols %>% select(CellType, Genes, Organism, Cell_Ontology_Class, DOI)
output
write_csv(output, path="reprogramming_analysis_table.csv")


```

# Scoring cell types by TF coexpression
```{r}
known.protocols <- read_csv("reprogramming_analysis_table.csv")
known.protocols <- known.protocols %>% filter(!is.na(Cell_Ontology_Class))
fractions <- read.csv("fractions.csv", row.names = 1)
tfs.used <- capitalize(tolower(unlist(lapply(as.character(known.protocols$Genes), strsplit, split=";"))))

fractions <- fractions[tfs.used, ]

```


```{r}
# First, divide each gene/celltype fraction by the maximum fraction per gene OR 0.1, whichever is larger.
max.fractions <- apply(fractions, 1, max)
max.fractions[max.fractions < 0.1] <- 0.1
fractions.scaled <- fractions / max.fractions

known.protocols$Rank_of_Expected_Type <- NA
# Annotate known reprogramming protocols with rank of "expected" cell type. In the case of multiple expected cell types, use the rank of the highest cell type.
Genesets <- character(0)
for(i in 1:nrow(known.protocols)) {
  
  Geneset <- known.protocols$Genes[i]
  Geneset <- gsub(pattern = ";",replacement = "__", x=Geneset)
  Genesets <- c(Geneset,Genesets)
  
  tfs <- capitalize(tolower(strsplit(Geneset, split="__")[[1]]))
  
  coexpr.score <- 0*fractions.scaled[1,] + 1
  for(tf in tfs){
    coexpr.score <- coexpr.score * fractions.scaled[tf,]
  }
  coexpr.ranks <- rank(-coexpr.score) # highest expressor will have lowest rank
  
  if(i==1) {
    all.coexpr.ranks <- coexpr.ranks
    all.coexpr.score <- coexpr.score
  } else{
    all.coexpr.ranks <- rbind(all.coexpr.ranks, coexpr.ranks)
    all.coexpr.score <- rbind(all.coexpr.score, coexpr.score)

  }

  celltypes.expected <- strsplit(as.character(known.protocols$Cell_Ontology_Class[i]), split = ";")[[1]]
  
  # get all tiss__cell_ontology_class combinations that match the expected cell_ontology_class
  # e.g. endothelial.cell is in multiple tissues and thus has multiple annotations
  all.annots <- as.character(tm.facs.metadata %>% filter(cell_ontology_class %in% celltypes.expected) %>% distinct(annotation.2) %>% pull(annotation.2))
  
  # get minimum rank of expected tissue-cell ontology combos
  rank.expected=min(coexpr.ranks[names(coexpr.ranks) %in% all.annots])
  known.protocols$Rank_of_Expected_Type[i] <- rank.expected
}
rownames(all.coexpr.ranks) <- Genesets
rownames(all.coexpr.score) <- Genesets

known.protocols <- known.protocols %>% select(CellType, Genes, Organism, Rank_of_Expected_Type, DOI)
write_csv(known.protocols, path = "reprogramming_analysis_table.csv")

```






