\clearpage
\section{Pancreas FACS}

\subsection{All Cells, labeled by \emph{Cell Ontology Class}}
\subsubsection{Table of cell counts in all cells, per \emph{Cell Ontology Class}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Cell Ontology Class}& Number of cells \\ \midrule
endothelial cell & 66 \\

leukocyte & 54 \\

pancreatic A cell & 390 \\

pancreatic D cell & 140 \\

pancreatic PP cell & 73 \\

pancreatic acinar cell & 182 \\

pancreatic ductal cell & 161 \\

pancreatic stellate cell & 49 \\

type B pancreatic cell & 449 \\
\bottomrule
\end{tabular}
\caption{Cell counts for all cells, per \emph{Cell Ontology Class}.}
\end{table}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_tsneplot.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_tsneplot_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. Bottom, legend mapping \emph{Cell Ontology Class} (and letter abbreviation) to colors}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (1 of 3, \emph{Amy2b}--\emph{Isl1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_1-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_1-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (1 of 3)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (2 of 3, \emph{Kdr}--\emph{Prss53})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_2-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_2-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (2 of 3)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (3 of 3, \emph{Ptf1a}--\emph{Sst})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_3-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_3-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (3 of 3)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (Raw Neurog3)}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_raw_Neurog3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_raw_Neurog3.pdf}
>>>>>>> autogenrate tex

\caption{\emph{Neurog3} transcripts are detected in islet, a few alpha, beta, and the majority of delta cells.
}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (Scaled Neurog3)}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_scaled_Neurog3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_scaled_Neurog3.pdf}
>>>>>>> autogenrate tex

\caption{\emph{Neurog3} transcripts are detected in islet, a few alpha, beta, and the majority of delta cells.
}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (Scaled Ppy)}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_scaled_Ppy.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_ridgeplot_scaled_Ppy.pdf}
>>>>>>> autogenrate tex

\caption{The transcript of PP cell signature hormone, \emph{Ppy}, is detected in delta cells at high levels.
}
\end{figure}


\clearpage

\subsubsection{Dotplot (1 of 3, \emph{Amy2b}--\emph{Isl1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_dotplot_1-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_dotplot_1-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (1 of 3)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (2 of 3, \emph{Kdr}--\emph{Prss53})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_dotplot_2-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_dotplot_2-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (2 of 3)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (3 of 3, \emph{Ptf1a}--\emph{Sst})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cell_ontology_class_dotplot_3-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cell_ontology_class_dotplot_3-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (3 of 3)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in all cells of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsection{All Cells, labeled by \emph{Cluster IDs}}
\subsubsection{Table of cell counts in all cells, per \emph{Cluster IDs}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Cluster IDs}& Number of cells \\ \midrule
0 & 342 \\

1 & 261 \\

2 & 207 \\

3 & 182 \\

4 & 161 \\

5 & 135 \\

6 & 107 \\

7 & 66 \\

8 & 54 \\

9 & 49 \\
\bottomrule
\end{tabular}
\caption{Cell counts for all cells, per \emph{Cluster IDs}.}
\end{table}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_tsneplot.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_tsneplot_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Cluster IDs} labels in all cells of Pancreas FACS. Bottom, legend mapping \emph{Cluster IDs} to colors}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (1 of 3, \emph{Amy2b}--\emph{Isl1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_ridgeplot_1-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cluster-ids_ridgeplot_1-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (1 of 3)  showing gene expression enrichment in \emph{Cluster IDs} labels in all cells of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Ridgeplot (2 of 3, \emph{Kdr}--\emph{Prss53})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_ridgeplot_2-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cluster-ids_ridgeplot_2-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (2 of 3)  showing gene expression enrichment in \emph{Cluster IDs} labels in all cells of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Ridgeplot (3 of 3, \emph{Ptf1a}--\emph{Sst})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_ridgeplot_3-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cluster-ids_ridgeplot_3-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (3 of 3)  showing gene expression enrichment in \emph{Cluster IDs} labels in all cells of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Dotplot (1 of 3, \emph{Amy2b}--\emph{Isl1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_dotplot_1-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cluster-ids_dotplot_1-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (1 of 3)  showing gene expression enrichment in \emph{Cluster IDs} labels in all cells of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Dotplot (2 of 3, \emph{Kdr}--\emph{Prss53})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_dotplot_2-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cluster-ids_dotplot_2-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (2 of 3)  showing gene expression enrichment in \emph{Cluster IDs} labels in all cells of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Dotplot (3 of 3, \emph{Ptf1a}--\emph{Sst})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_cluster-ids_dotplot_3-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_cluster-ids_dotplot_3-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (3 of 3)  showing gene expression enrichment in \emph{Cluster IDs} labels in all cells of Pancreas FACS. }
\end{figure}


\clearpage

\subsection{All Cells, labeled by \emph{Free Annotation}}
\subsubsection{Table of cell counts in all cells, per \emph{Free Annotation}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Free Annotation}& Number of cells \\ \midrule
acinar cell & 182 \\

beta cell & 449 \\

ductal cell & 161 \\

endothelial cell & 66 \\

immune cell & 54 \\

pancreatic A cell & 390 \\

pancreatic D cell & 140 \\

pancreatic PP cell & 73 \\

stellate cell & 49 \\
\bottomrule
\end{tabular}
\caption{Cell counts for all cells, per \emph{Free Annotation}.}
\end{table}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_tsneplot.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_tsneplot_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Free Annotation} labels in all cells of Pancreas FACS. Bottom, legend mapping \emph{Free Annotation} (and letter abbreviation) to colors}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (1 of 3, \emph{Amy2b}--\emph{Isl1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_ridgeplot_1-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_free_annotation_ridgeplot_1-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (1 of 3)  showing gene expression enrichment in \emph{Free Annotation} labels in all cells of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (2 of 3, \emph{Kdr}--\emph{Prss53})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_ridgeplot_2-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_free_annotation_ridgeplot_2-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (2 of 3)  showing gene expression enrichment in \emph{Free Annotation} labels in all cells of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (3 of 3, \emph{Ptf1a}--\emph{Sst})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_ridgeplot_3-of-3.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/allcells_free_annotation_ridgeplot_3-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (3 of 3)  showing gene expression enrichment in \emph{Free Annotation} labels in all cells of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (1 of 3, \emph{Amy2b}--\emph{Isl1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_dotplot_1-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_free_annotation_dotplot_1-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (1 of 3)  showing gene expression enrichment in \emph{Free Annotation} labels in all cells of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (2 of 3, \emph{Kdr}--\emph{Prss53})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_dotplot_2-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_free_annotation_dotplot_2-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (2 of 3)  showing gene expression enrichment in \emph{Free Annotation} labels in all cells of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (3 of 3, \emph{Ptf1a}--\emph{Sst})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/allcells_free_annotation_dotplot_3-of-3.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/allcells_free_annotation_dotplot_3-of-3.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (3 of 3)  showing gene expression enrichment in \emph{Free Annotation} labels in all cells of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsection{Delta Cells, labeled by \emph{Expression}}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/DeltaCells_expression_tsneplot.pdf}

\caption{\emph{Prss53} transcripts are only detected in islet beta cells.
}
\end{figure}


\clearpage

\subsection{Delta Cells, labeled by \emph{Mouse Id}}
\subsubsection{Table of cell counts in delta cells, per \emph{Mouse Id}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Mouse Id}& Number of cells \\ \midrule\bottomrule
\end{tabular}
\caption{Cell counts for delta cells, per \emph{Mouse Id}.}
\end{table}

\clearpage
<<<<<<< 0934876587e0a84d261fc48e5d7a4644ec19d4e0
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\subsubsection{Scatterplot (Raw \emph{Neurog3} \texttimes \emph{Hhex})}
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/DeltaCells_mouse-id_scatterplot_raw_Neurog3-x-Hhex.pdf}
=======
\subsubsection{Scatterplot (Raw \emph{Neurog3} $\times$ \emph{Hhex})}
=======
\subsubsection{Scatterplot (Raw \emph{Neurog3} \texttimes \emph{Hhex})}
>>>>>>> autogenerate tex
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/DeltaCells_mouse-id_scatterplot_raw_Neurog3-x-Hhex.pdf}
>>>>>>> autogenrate tex

\caption{Delta heterogeneity is observed in every individual mouse.
}
\end{figure}


\clearpage

<<<<<<< 0934876587e0a84d261fc48e5d7a4644ec19d4e0
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\subsubsection{Scatterplot (Scaled \emph{Neurog3} \texttimes \emph{Hhex})}
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/DeltaCells_mouse-id_scatterplot_scaled_Neurog3-x-Hhex.pdf}
=======
\subsubsection{Scatterplot (Scaled \emph{Neurog3} $\times$ \emph{Hhex})}
=======
\subsubsection{Scatterplot (Scaled \emph{Neurog3} \texttimes \emph{Hhex})}
>>>>>>> autogenerate tex
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/DeltaCells_mouse-id_scatterplot_scaled_Neurog3-x-Hhex.pdf}
>>>>>>> autogenrate tex

\caption{Delta heterogeneity is observed in every individual mouse.
}
\end{figure}


\clearpage

<<<<<<< 0934876587e0a84d261fc48e5d7a4644ec19d4e0
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\subsection{Delta Cells, labeled by \emph{Neurog3} \textgreater 0}

\clearpage
\subsubsection{Scatterplot (Raw \emph{Neurog3} \texttimes \emph{Hhex})}
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/DeltaCells_Neurog3>0_scatterplot_raw_Neurog3-x-Hhex.pdf}

\caption{Scatter plot of all delta cells reveals that \emph{Neurog3} and \emph{Hhex} mark three distinct groups of
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}$^\text{neg}$, \emph{Neurog3}$^\text{neg}$\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
=======
\subsection{Delta Cells, labeled by \emph{Neurog3} $> 0$}
=======
\subsection{Delta Cells, labeled by \emph{Neurog3} \textgreater 0}
>>>>>>> autogenerate tex

\clearpage
\subsubsection{Scatterplot (Raw \emph{Neurog3} \texttimes \emph{Hhex})}
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/DeltaCells_Neurog3>0_scatterplot_raw_Neurog3-x-Hhex.pdf}

\caption{Scatter plot of all delta cells reveals that \emph{Neurog3} and \emph{Hhex} mark three distinct groups of
<<<<<<< 224670a25d7e71f8625110ad1c3440285d60ed49
<<<<<<< 0934876587e0a84d261fc48e5d7a4644ec19d4e0
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}^{neg}, \emph{Neurog3}^{neg}\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
>>>>>>> autogenrate tex
=======
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}$^\text{neg}$, \emph{Neurog3}$$^\text{neg}$$\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
>>>>>>> autogenerate tex
=======
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}$^\text{neg}$, \emph{Neurog3}$^\text{neg}$\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
>>>>>>> Autogenerate tex
}
\end{figure}


\clearpage

<<<<<<< 0934876587e0a84d261fc48e5d7a4644ec19d4e0
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\subsubsection{Scatterplot (Scaled \emph{Neurog3} \texttimes \emph{Hhex})}
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/DeltaCells_Neurog3>0_scatterplot_scaled_Neurog3-x-Hhex.pdf}

\caption{Scatter plot of all delta cells reveals that \emph{Neurog3} and \emph{Hhex} mark three distinct groups of
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}$^\text{neg}$, \emph{Neurog3}$^\text{neg}$\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
=======
\subsubsection{Scatterplot (Scaled \emph{Neurog3} $\times$ \emph{Hhex})}
=======
\subsubsection{Scatterplot (Scaled \emph{Neurog3} \texttimes \emph{Hhex})}
>>>>>>> autogenerate tex
\begin{figure}[h]
\centering
\includegraphics[width=.6\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/DeltaCells_Neurog3>0_scatterplot_scaled_Neurog3-x-Hhex.pdf}

\caption{Scatter plot of all delta cells reveals that \emph{Neurog3} and \emph{Hhex} mark three distinct groups of
<<<<<<< 0934876587e0a84d261fc48e5d7a4644ec19d4e0
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}^{neg}, \emph{Neurog3}^{neg}\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
>>>>>>> autogenrate tex
=======
delta cells, i.e. \emph{Neurog3}$^+$\emph{Hhex}$^\text{neg}$, \emph{Neurog3}$^\text{neg}$\emph{Hhex}$^+$, and \emph{Neurog3}$^+$\emph{Hhex}$^+$.
>>>>>>> autogenerate tex
}
\end{figure}


\clearpage
\subsection{Subset A, highlighted from All Cells tSNE}
\subsubsection{t-SNE plot (Allcells)}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_highlighted_tsneplot_allcells.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_highlighted_tsneplot_allcells_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Highlighted} labels in Subset A of Pancreas FACS. Bottom, legend mapping \emph{Highlighted} (and letter abbreviation) to colors}
\end{figure}


\clearpage

\subsection{Subset A, labeled by \emph{Cell Ontology Class}}
\subsubsection{Table of cell counts in Subset A, per \emph{Cell Ontology Class}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Cell Ontology Class}& Number of cells \\ \midrule
pancreatic A cell & 390 \\

pancreatic D cell & 140 \\

pancreatic PP cell & 73 \\
\bottomrule
\end{tabular}
\caption{Cell counts for Subset A, per \emph{Cell Ontology Class}.}
\end{table}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cell_ontology_class_tsneplot.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cell_ontology_class_tsneplot_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Cell Ontology Class} labels in Subset A of Pancreas FACS. Bottom, legend mapping \emph{Cell Ontology Class} (and letter abbreviation) to colors}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (1 of 2, \emph{Ang}--\emph{Irx1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cell_ontology_class_ridgeplot_1-of-2.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cell_ontology_class_ridgeplot_1-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (1 of 2)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in Subset A of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (2 of 2, \emph{Isl1}--\emph{Vsig1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cell_ontology_class_ridgeplot_2-of-2.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cell_ontology_class_ridgeplot_2-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (2 of 2)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in Subset A of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (1 of 2, \emph{Ang}--\emph{Irx1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cell_ontology_class_dotplot_1-of-2.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cell_ontology_class_dotplot_1-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (1 of 2)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in Subset A of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (2 of 2, \emph{Isl1}--\emph{Vsig1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cell_ontology_class_dotplot_2-of-2.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cell_ontology_class_dotplot_2-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (2 of 2)  showing gene expression enrichment in \emph{Cell Ontology Class} labels in Subset A of Pancreas FACS. A: endothelial cell, B: leukocyte, C: pancreatic A cell, D: pancreatic acinar cell, E: pancreatic D cell, F: pancreatic ductal cell, G: pancreatic PP cell, H: pancreatic stellate cell, I: type B pancreatic cell.}
\end{figure}


\clearpage

\subsection{Subset A, labeled by \emph{Cluster IDs}}
\subsubsection{Table of cell counts in Subset A, per \emph{Cluster IDs}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Cluster IDs}& Number of cells \\ \midrule
1 & 261 \\

2 & 207 \\

5 & 135 \\
\bottomrule
\end{tabular}
\caption{Cell counts for Subset A, per \emph{Cluster IDs}.}
\end{table}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cluster-ids_tsneplot.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cluster-ids_tsneplot_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Cluster IDs} labels in Subset A of Pancreas FACS. Bottom, legend mapping \emph{Cluster IDs} to colors}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (1 of 2, \emph{Ang}--\emph{Irx1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cluster-ids_ridgeplot_1-of-2.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cluster-ids_ridgeplot_1-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (1 of 2)  showing gene expression enrichment in \emph{Cluster IDs} labels in Subset A of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Ridgeplot (2 of 2, \emph{Isl1}--\emph{Vsig1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cluster-ids_ridgeplot_2-of-2.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cluster-ids_ridgeplot_2-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (2 of 2)  showing gene expression enrichment in \emph{Cluster IDs} labels in Subset A of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Dotplot (1 of 2, \emph{Ang}--\emph{Irx1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cluster-ids_dotplot_1-of-2.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cluster-ids_dotplot_1-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (1 of 2)  showing gene expression enrichment in \emph{Cluster IDs} labels in Subset A of Pancreas FACS. }
\end{figure}


\clearpage

\subsubsection{Dotplot (2 of 2, \emph{Isl1}--\emph{Vsig1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_cluster-ids_dotplot_2-of-2.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_cluster-ids_dotplot_2-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (2 of 2)  showing gene expression enrichment in \emph{Cluster IDs} labels in Subset A of Pancreas FACS. }
\end{figure}


\clearpage

\subsection{Subset A, labeled by \emph{Free Annotation}}
\subsubsection{Table of cell counts in Subset A, per \emph{Free Annotation}}\begin{table}[h]
\centering
\label{my-label}
\begin{tabular}{@{}ll@{}}
\toprule

\emph{Free Annotation}& Number of cells \\ \midrule
pancreatic A cell & 390 \\

pancreatic D cell & 140 \\

pancreatic PP cell & 73 \\
\bottomrule
\end{tabular}
\caption{Cell counts for Subset A, per \emph{Free Annotation}.}
\end{table}

\clearpage
\subsubsection{t-SNE plot}
\begin{figure}[h]
\centering
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_free_annotation_tsneplot.pdf}
\includegraphics[height=.35\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_free_annotation_tsneplot_legend.pdf}
\caption{Top, t-Distributed stochastic neighbor embedding (tSNE) plot  \emph{Free Annotation} labels in Subset A of Pancreas FACS. Bottom, legend mapping \emph{Free Annotation} (and letter abbreviation) to colors}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (1 of 2, \emph{Ang}--\emph{Irx1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_free_annotation_ridgeplot_1-of-2.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_free_annotation_ridgeplot_1-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (1 of 2)  showing gene expression enrichment in \emph{Free Annotation} labels in Subset A of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Ridgeplot (2 of 2, \emph{Isl1}--\emph{Vsig1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[width=.65\textwidth]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_free_annotation_ridgeplot_2-of-2.pdf}
=======
\includegraphics[width=.65\textwidth]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_free_annotation_ridgeplot_2-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Ridgeplot (2 of 2)  showing gene expression enrichment in \emph{Free Annotation} labels in Subset A of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (1 of 2, \emph{Ang}--\emph{Irx1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_free_annotation_dotplot_1-of-2.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_free_annotation_dotplot_1-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (1 of 2)  showing gene expression enrichment in \emph{Free Annotation} labels in Subset A of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}


\clearpage

\subsubsection{Dotplot (2 of 2, \emph{Isl1}--\emph{Vsig1})}
\begin{figure}[h]
\centering
<<<<<<< 9dfdae4f73f742e098de608ea0999c2476425095
\includegraphics[angle=90, height=.65\textheight]{/Users/olgabot/tabula-muris-supplemental/Pancreas/facs/SubsetA_free_annotation_dotplot_2-of-2.pdf}
=======
\includegraphics[angle=90, height=.65\textheight]{../30_tissue_supplement_figures/Pancreas/facs/SubsetA_free_annotation_dotplot_2-of-2.pdf}
>>>>>>> autogenrate tex

\caption{ Dotplot (2 of 2)  showing gene expression enrichment in \emph{Free Annotation} labels in Subset A of Pancreas FACS. A: acinar cell, B: beta cell, C: ductal cell, D: endothelial cell, E: immune cell, F: pancreatic A cell, G: pancreatic D cell, H: pancreatic PP cell, I: stellate cell.}
\end{figure}

